SimGeneral/MixingModule/python/caloTruthProducer_cfi.py

```diff
@@ -59,3 +53,20 @@
 from Configuration.Eras.Modifier_fastSim_cff import fastSim
 fastSim.toReplaceWith(caloParticles, cms.PSet()) # don't allow this to run in fastsim
 
+from Configuration.Eras.Modifier_phase2_common_cff import phase2_common
+phase2_common.toModify(
```

Comment:
Rather than tying everything to `phase2_common`, how about doing the HGCal-specific customizations with `phase2_hgcal` modifier, that is used e.g. in
https://github.com/cms-sw/cmssw/blob/df4abb9f42214557f727db8ff152cedb369eedfa/SimGeneral/MixingModule/python/digitizers_cfi.py#L57-L63
and 
https://github.com/cms-sw/cmssw/blob/df4abb9f42214557f727db8ff152cedb369eedfa/SimGeneral/PreMixingModule/python/mixOne_premix_on_sim_cfi.py#L274-L282
?

Reply:
In addition (although this may strictly speaking be outside of the scope of this PR), the current setup of the Modifiers is confusing
* by default the `hgc` collections are used
* `phase2_hfnose` adds one collection to the `hgc` set
* `run3_ecalclustering` replaces collections to `ecal` ones
* `fastsim` disables CaloParticles altogether
* `phase2_common` sets the collections to `hgc`, `ecal`, and `ecal`

To me a clearer setup would be something along
* default is to have empty `simHitCollections = cms.PSet()`
* `run3_ecalclustering` sets the 3 `ecal` collections (EB, EE, ES)
* `phase2_common` sets the `ecal` and `hcal` collections
  * Most consistent (currently, wrt `mixOne_premix_on_sim_cfi.py`) would be to include both EB and EE in `ecal`, i.e. remove ES, at this stage
  * Now that also `phase2_ecal` and `phase2_hcal` Modifiers exist, perhaps it would make sense to use those instead of `phase2_common` in all the *MixingModule configurations? (needs to be done consistently)
* `phase2_hgcal` sets the `ecal` to EB (or removes EE), and adds the `hgc`
* `fastsim` disables CaloParticles altogether